apollo-testing-support/src/commonMain/kotlin/com/apollographql/apollo3/testing/ApolloMockDispatcher.kt

```diff
@@ -0,0 +1,34 @@
+package com.apollographql.apollo3.testing
+
+import com.apollographql.apollo3.annotations.ApolloExperimental
+import com.apollographql.apollo3.api.ApolloResponse
+import com.apollographql.apollo3.api.CustomScalarAdapters
+import com.apollographql.apollo3.api.Operation
+import com.apollographql.apollo3.api.composeJsonResponse
+import com.apollographql.apollo3.api.json.buildJsonString
+import com.apollographql.apollo3.mockserver.MockDispatcher
+import com.apollographql.apollo3.mockserver.MockRecordedRequest
+import com.apollographql.apollo3.mockserver.MockResponse
+import com.apollographql.apollo3.mockserver.MockServer
+
+interface ApolloMockDispatcher {
+  val customScalarAdapters: CustomScalarAdapters
+  fun dispatch(request: MockRecordedRequest): ApolloResponse<out Operation.Data>
```

Comment:
I'm still digging into the diff but I agree with your general sentiment that this is not really needed as there is:

* `MockServer` for testing at the HTTP level
* `TestNetworkTransport` for testing at the GraphQL level

Reply:
Yeah that seems sensible. If you're ok I'll axe this part (no regrets, at least I learned a few things doing it!).